torch/utils/data/dataloader.py

```diff
@@ -330,6 +377,12 @@ def _shutdown_workers(self):
                     pass
                 for q in self.index_queues:
                     q.put(None)
+                # We should wait for the workers to finish before shutting down
+                # the manager process, otherwise in the `spawn` start method
+                # workers won't be able to access the queues from manager
+                # and will give FileNotFoundError (https://bugs.python.org/issue28965)
+                for w in self.workers:
+                    w.join()
```

Comment:
A `join` with no timeout doesn't seem like a good idea for a function called from `__del__`. What's the issue with the workers? If they exit because of an error, which is caused by the master process then great, they won't stay alive!